modules/mod_ginger_collection/lib/js/search/components/filters-subjects.js

```diff
@@ -8,11 +8,33 @@ $.widget('ui.search_cmp_filters_subjects', {
         me.widgetElement = widgetElement;
         me.type = 'subject';
 
+        this.loadMore();
+
         inputs.on('change', function () {
             $(document).trigger('search:inputChanged');
         });
     },
 
+    loadMore: function() {
+        var me = this,
+            btn = $('.filter-down-btn'),
+            listItem = me.widgetElement.find('li.rest-subject'),
+            counter = 1;
+
+        btn.on('click', function () {
+            counter = counter + 10;
+            $.each(listItem, function(i , val){
+                if(i < counter) {
+                    $(val).removeClass('hidden');
+                }
+            });
+
+            if(counter + counter > listItem.length) {
+                $(this).hide();
+            }
+        });
+    },
```

Comment:
This makes the ‘load more’ functionality dependent on this particular filter. That seems wrong to me: it’s generic functionality that is applicable to all filters.